src/main/charts/confluence/templates/ingress-setup.yaml

```diff
@@ -0,0 +1,37 @@
+{{- if .Values.ingress.create }}
+{{/* Change to networking.k8s.io/v1 when k8s 1.19 is the oldest supported version */}}
+apiVersion: networking.k8s.io/v1beta1
+kind: Ingress
+metadata:
+  name: {{ include "confluence.fullname" . }}-setup
+  labels:
+    {{- include "confluence.labels" . | nindent 4 }}
+  annotations:
+    {{ if .Values.ingress.nginx }}
+    "kubernetes.io/ingress.class": "nginx"
+    "nginx.ingress.kubernetes.io/affinity": "cookie"
+    "nginx.ingress.kubernetes.io/affinity-mode": "persistent"
+    "nginx.ingress.kubernetes.io/proxy-body-size": {{ .Values.ingress.maxBodySize }}
+    "nginx.ingress.kubernetes.io/proxy-connect-timeout": "300"
```

Comment:
This is the separate Ingress with increasing timeout. It's existing for `/setup` endpoint for Confluence, because those HTTP requests are expected to be slow

Reply:
I am hesitant to introduce another `Ingress` just for the purpose of `setup` slow endpoints. This will stay there even after the setup was finished and it might be just confusing for customers. But tbh I didn't find a better workaround for now. Clearly the best solution is to provide fully automated setup in Confluence but that is a long term solution. I was looking into `Jobs` and Helm `hooks`. None will solve our use case to manipulate the values of a resources on certain conditions that will happen after manual intervention. Operator could actually do that but again that is a bigger project.

For now, we can probably at least stick with DRY. We can create `_ingress-template.yaml` file that will contain the current content with the flexibility to change the name and the timeout. That will drastically simplify the two ingresses (because they are mostly duplicated). More information about library charts is here: https://helm.sh/docs/topics/library_charts/

Reply:
I've extracted the ingress to the library `_ingress.tpl` and trimmed `ingress.yaml` and `ingress-setup.yaml` to define only parts which are differ.
The repetition is gone. I can't say I like it very much though, because using libraries in Helm is pretty ugly. You have to use a common chart utility to merge 2 trees together, and this adds complexity.
One more thing - because this common merge utility is used, Confluence chart right now has a dependency on a common library. I had to add some code to tests and `helm_install.sh` script which updates the dependency prior to packaging (it will now work otherwire).
All in all, I think it should be OK in the current state. Unit tests are okay and I was also able to provision and test Confluence. Let's see how it goes.

Reply:
I am really sorry for flip-flopping on this one. I think the original intent was to keep it simple and seems that `helm` is not really providing us with the tools to DRY. Using the library is probably adding the same amount of code as removes but also increases complexity and dependency on another library, which is deprecated.

From my side, I would then rather prefer to stick with two very similar ingresses than trying to refactor `yaml` using an external library. Hopefully, this shouldn't be a big change as we have the files already in the git history and we can just fall back to the previous solution.

Reply:
I've reverted those commits and now we are back to the old solution